### ISSUE TITLE
Editor crashes when adding a ColorRect node to an existing scene

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#55014]
### Godot version

3.5.beta.custom_build (458f5b42a63ca0f9b2199e21165354a1e9938e37)

### System information

Fedora 34, GeForce GTX 1080 (NVIDIA 470.74)

### Issue description

The editor crashes when adding a ColorRect node to an existing scene. This only occurs on `3.x`, not `master` or 3.4-stable.

Backtrace:

```
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v3.5.beta.custom_build (458f5b42a63ca0f9b2199e21165354a1e9938e37)
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib64/libc.so.6(+0x3d320) [0x7f0f8c975320] (??:0)
[2] bin/godot.x11.tools.64.llvm(__dynamic_cast+0x1d) [0x673e4cd] (??:?)
[3] Reference* Object::cast_to<Reference>(Object*) (/home/hugo/Documents/Git/godotengine/godot/./core/object.h:602)
[4] Variant::Variant(Object const*) (/home/hugo/Documents/Git/godotengine/godot/core/variant.cpp:2163)
[5] SceneTreeDock::_do_create(Node*) (/home/hugo/Documents/Git/godotengine/godot/editor/scene_tree_dock.cpp:2160)
[6] SceneTreeDock::_create() (/home/hugo/Documents/Git/godotengine/godot/editor/scene_tree_dock.cpp:2214)
[7] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (/home/hugo/Documents/Git/godotengine/godot/./core/method_bind.gen.inc:59)
[8] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:918)
[9] Object::emit_signal(StringName const&, Variant const**, int) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:1224)
[10] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:1279)
[11] CreateDialog::_confirmed() (/home/hugo/Documents/Git/godotengine/godot/editor/create_dialog.cpp:452)
[12] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (/home/hugo/Documents/Git/godotengine/godot/./core/method_bind.gen.inc:59)
[13] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:918)
[14] Object::emit_signal(StringName const&, Variant const**, int) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:1224)
[15] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:1279)
[16] AcceptDialog::_ok_pressed() (/home/hugo/Documents/Git/godotengine/godot/scene/gui/dialogs.cpp:398)
[17] AcceptDialog::_builtin_text_entered(String const&) (/home/hugo/Documents/Git/godotengine/godot/scene/gui/dialogs.cpp:391)
[18] MethodBind1<String const&>::call(Object*, Variant const**, int, Variant::CallError&) (/home/hugo/Documents/Git/godotengine/godot/./core/method_bind.gen.inc:775)
[19] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:918)
[20] Object::emit_signal(StringName const&, Variant const**, int) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:1224)
[21] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:1279)
[22] LineEdit::_gui_input(Ref<InputEvent>) (/home/hugo/Documents/Git/godotengine/godot/scene/gui/line_edit.cpp:328)
[23] MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) (/home/hugo/Documents/Git/godotengine/godot/./core/method_bind.gen.inc:775)
[24] Object::call_multilevel(StringName const&, Variant const**, int) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:766)
[25] Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:864)
[26] Viewport::_gui_input_event(Ref<InputEvent>) (/home/hugo/Documents/Git/godotengine/godot/scene/main/viewport.cpp:2395)
[27] Viewport::input(Ref<InputEvent> const&) (/home/hugo/Documents/Git/godotengine/godot/scene/main/viewport.cpp:2810)
[28] Viewport::_vp_input(Ref<InputEvent> const&) (/home/hugo/Documents/Git/godotengine/godot/scene/main/viewport.cpp:1422)
[29] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (/home/hugo/Documents/Git/godotengine/godot/./core/method_bind.gen.inc:775)
[30] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:918)
[31] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:848)
[32] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/hugo/Documents/Git/godotengine/godot/scene/main/scene_tree.cpp:272)
[33] SceneTree::input_event(Ref<InputEvent> const&) (/home/hugo/Documents/Git/godotengine/godot/scene/main/scene_tree.cpp:431)
[34] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (/home/hugo/Documents/Git/godotengine/godot/main/input_default.cpp:477)
[35] InputDefault::flush_buffered_events() (/home/hugo/Documents/Git/godotengine/godot/main/input_default.cpp:698)
[36] OS_X11::process_xevents() (/home/hugo/Documents/Git/godotengine/godot/platform/x11/os_x11.cpp:2968)
[37] OS_X11::run() (/home/hugo/Documents/Git/godotengine/godot/platform/x11/os_x11.cpp:3701)
[38] bin/godot.x11.tools.64.llvm(main+0x173) [0x2de5ec3] (/home/hugo/Documents/Git/godotengine/godot/platform/x11/godot_x11.cpp:55)
[39] /lib64/libc.so.6(__libc_start_main+0xd5) [0x7f0f8c95fb75] (??:0)
[40] bin/godot.x11.tools.64.llvm(_start+0x2e) [0x2de5c8e] (??:?)
```

### Steps to reproduce

- Add a ColorRect node to an existing 3D scene.

### Minimal reproduction project

Since an existing project is needed as [new scenes can't be created currently](https://github.com/godotengine/godot/issues/55015): [test_giprobe_bake_runtime_1.zip](https://github.com/godotengine/godot/files/7542335/test_giprobe_bake_runtime_1.zip)

**Comment by akien-mga** [comment]
Same test results as https://github.com/godotengine/godot/issues/55015#issuecomment-969955386, this is likely a regression from #54433.

**Comment by akien-mga** [comment]
Fixed by #55022.
